### DISCUSSION TITLE
sahi with SSD or Yolov4/7??

### DISCUSSION THREAD

**Discussion by Rushikesh-pawar** [obss/sahi#685] [Q&A] [answered]
Can we use sahi with other object detection models as well? for eg yolov4/7, SSD etc?

**Comment by kadirnar** [comment, marked as answer]
@Rushikesh-pawar All you need to do is, creating a new class in [sahi/models](https://github.com/obss/sahi/blob/main/sahi/models/) folder that implements [DetectionModel class](https://github.com/obss/sahi/blob/21ecb285aa6bf93c2a00605dfb9b138f19d8d62d/sahi/model.py#L21).

Yolov7: https://github.com/kadirnar/Yolov7-SAHI
Yolov6: https://github.com/kadirnar/Yolov6-SAHI
YOLOX: https://github.com/obss/sahi/pull/557
TensorFlowHub: https://github.com/obss/sahi/pull/501

**Reply by saholo21** [reply]
Hi @kadirnar , I checked out your Yolov7+SAHI colab link but it doesn't work

**Reply by kadirnar** [reply]
It works locally. Can you show the error message?

**Reply by saholo21** [reply]
Sorry, it was just a github authorization issue, my bad

**Reply by Rohan-Python** [reply]
Hey can you please tell how  to recreate class for yolov7. I did not understood it. 😐

**Reply by kadirnar** [reply]
Yolov7-Pip: https://github.com/kadirnar/Yolov7-SAHI/blob/main/sahi/models/yolov7pip.py

Yolov7Hub: https://github.com/kadirnar/Yolov7-SAHI/blob/main/sahi/models/yolov7hub.py

Demo: https://github.com/kadirnar/Yolov7-SAHI/blob/main/demo/yolov7_demo.py

**Comment by Rohan-Python** [comment]
can you please help me a bit . I dont know what to do ahead. How do i get
to use yolov7 with sahi

**Reply by kadirnar** [reply]
@Rohan-Python ,
Yolov7-SAHI:
```
pip install yolov7detect
git clone https://github.com/kadirnar/Yolov7-SAHI
cd Yolov7-SAHI
```
Create a new python file and write the following codes:
```
from sahi.predict import get_sliced_prediction, predict, get_prediction
from sahi import AutoDetectionModel

# yolov7 model path
yolov7_model_path = 'models/yolov7.pt'

# image path
image_path =  "demo_data/small-vehicles1.jpeg"

# Detection
detection_model = AutoDetectionModel.from_pretrained(
    model_type='yolov7pip', # or 'yolov7hub'
    model_path=yolov7_model_path,
    confidence_threshold=0.3,
    device="cpu", # or 'cuda:0'
)

result = get_sliced_prediction(
    image_path,
    detection_model,
    slice_height = 256,
    slice_width = 256,
    overlap_height_ratio = 0.2,
    overlap_width_ratio = 0.2
)

result.export_visuals(export_dir="demo_data/")
```
Just update image_path and yolov7_model_path variables.

**Reply by Rohan-Python** [reply]
![image](https://user-images.githubusercontent.com/80620860/210185508-39a54128-85bf-4470-910e-b40397e43ae0.png)
I did Exactly as you said. But I am facing this error.
and 
Thank you for trying to help me out.. Means a lot

**Reply by fcakyon** [reply]
@Rohan-Python i guess you have to install the sahi fork as:
`pip uninstall sahi`
`pip install git+https://github.com/kadirnar/Yolov7-SAHI.git`

**Reply by kadirnar** [reply]
https://github.com/kadirnar/Yolov7-SAHI/blob/main/sahi/auto_model.py#L5

You should try "yolov7pip".

**Reply by Rohan-Python** [reply]
![image](https://user-images.githubusercontent.com/80620860/210185795-307bce15-01f4-4f22-a2c0-e9069c84d165.png)
same again

**Reply by Rohan-Python** [reply]
> @Rohan-Python i guess you have to install the sahi fork as: `pip uninstall sahi` `pip install git+https://github.com/kadirnar/Yolov7-SAHI.git`

Yeahh This worked for me.. Thank you man. I uninstalled sahi and forked it.

**Reply by Rohan-Python** [reply]
Can we not use --source as 0   to take real time video as input to run yolo object detection with sahi ?

**Reply by kadirnar** [reply]
> Can we not use --source as 0 to take real time video as input to run yolo object detection with sahi ?

https://github.com/kadirnar/Yolov7-SAHI/blob/main/docs/cli.md
Example:
```
sahi predict --model_path yolov5s.pt --model_type yolov5 --source video.mp4 --view_video
```
Add the ```view_video``` parameter.
